cmd/medius/images/push.go

```diff
@@ -247,6 +236,77 @@ func (b *buildAndPublish) readArtifact(artifactReader http.ReadCloserWithChecksu
 	return file.Name(), nil
 }
 
+func (b *buildAndPublish) buildImages(entry *common.Entry) ([]v1.Image, []string, error) {
+	var images []v1.Image
+	var artifacts []string
+
+	rebuildNeeded, err := b.rebuildNeeded(entry)
+	if err != nil {
+		return nil, nil, err
+	}
+	if !rebuildNeeded && !b.Options.PublishImagesOptions.ForceBuild {
+		b.Log.Info("Nothing to do.")
+		return nil, nil, nil
+	}
+	if errors.Is(b.Ctx.Err(), context.Canceled) {
+		return nil, nil, b.Ctx.Err()
+	}
```

Comment:
To me it makes more sense to keep this out of this method and to place it in `Do`. It's not about building images but rather about deciding if we need to do it at all.

Reply:
Done.